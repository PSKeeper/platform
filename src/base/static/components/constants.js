export default {
  TEXT_FIELD_TYPENAME: "text",
  TEXTAREA_FIELD_TYPENAME: "textarea",
  RICH_TEXTAREA_FIELD_TYPENAME: "rich_textarea",
  BIG_CHECKBOX_FIELD_TYPENAME: "big_checkbox",
  BIG_RADIO_FIELD_TYPENAME: "big_radio",
  DROPDOWN_FIELD_TYPENAME: "dropdown",
  DROPDOWN_AUTOCOMPLETE_FIELD_TYPENAME: "dropdown_autocomplete",
  PUBLISH_CONTROL_TOOLBAR_TYPENAME: "publish_control_toolbar",
  MAP_DRAWING_TOOLBAR_TYPENAME: "map_drawing_toolbar",
  CUSTOM_URL_TOOLBAR_TYPENAME: "custom_url_toolbar",
  DATETIME_FIELD_TYPENAME: "datetime",
  GEOCODING_FIELD_TYPENAME: "geocoding",
  BIG_TOGGLE_FIELD_TYPENAME: "big_toggle",
  ATTACHMENT_FIELD_TYPENAME: "file",
  COMMON_FORM_ELEMENT_TYPENAME: "common_form_element",
  SUBMIT_FIELD_TYPENAME: "submit",
  RANGE_FIELD_TYPENAME: "range",

  SUBMITTER_FIELDNAME: "submitter_name",

  AUTOFILL_DURATION_DAYS: 30,

  LOCATION_TYPE_PROPERTY_NAME: "location_type",
  TITLE_PROPERTY_NAME: "title",
  FULL_TITLE_PROPERTY_NAME: "fullTitle",
  NAME_PROPERTY_NAME: "name",
  SUBMITTER_FIELD_NAME: "submitter",
  USER_TOKEN_PROPERTY_NAME: "user_token",
  CREATED_DATETIME_PROPERTY_NAME: "created_datetime",
  DATASET_SLUG_PROPERTY_NAME: "datasetSlug",
  MODEL_ID_PROPERTY_NAME: "id",
  STORY_FIELD_NAME: "story",
  DATASET_ID_PROPERTY_NAME: "datasetId",

  ATTACHMENT_TYPE_PROPERTY_NAME: "type",
  ATTACHMENT_NAME_PROPERTY_NAME: "name",
  ATTACHMENT_FILE_PROPERTY_NAME: "file",
  RICH_TEXT_IMAGE_CODE: "RT",
  COVER_IMAGE_CODE: "CO",

  RICH_TEXT_IMAGE_MARKUP_PREFIX: "{{#rich-text-image ",
  RICH_TEXT_IMAGE_MARKUP_SUFFIX: "}}",

  FIELD_STATE_VALUE_KEY: "value",
  FIELD_STATE_VALIDITY_KEY: "isValid",
  FIELD_STATE_VALIDITY_MESSAGE_KEY: "message",
  FIELD_STATE_FIELD_TYPE_KEY: "type",
  FIELD_STATE_RENDER_KEY: "renderKey",

  GEOMETRY_PROPERTY_NAME: "geometry",
  GEOMETRY_STYLE_PROPERTY_NAME: "style",

  ON_ADD_ATTACHMENT_ACTION: "on-add-attachment",

  GEOMETRY_TYPE_PROPERTY_NAME: "type",
  ICON_URL_PROPERTY_NAME: "iconUrl",
  GEOMETRY_TYPE_MAPPINGS: {
    LineString: "polyline",
    Polygon: "polygon",
    Point: "marker",
  },
  GEOMETRY_EDITOR_TOOL_MAPPINGS: {
    LineString: "edit-polyline",
    Polygon: "edit-polygon",
    Point: "edit-marker",
  },
};
