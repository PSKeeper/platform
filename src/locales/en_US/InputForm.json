{
  "validationHeader":
    "Your post is looking good, but we need some more information before we can share it on the map."
}
