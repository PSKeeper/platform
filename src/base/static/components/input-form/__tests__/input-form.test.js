import React from "react";
import { shallow } from "enzyme";
import InputForm from "../index.js";
import FormField from "../../form-fields/form-field";
import constants from "../../../constants";
import WarningMessagesContainer from "../../ui-elements/warning-messages-container";
import { OrderedMap, Map } from "immutable";

describe("InputForm", () => {
  // TODO: consider generalizing this stub into a mock:
  const categoryName = "someCategory";
  const eventStub = { preventDefault: () => {} };
  const defaultProps = {
    container: {},
    hideCenterPoint: () => {},
    hideSpotlightMask: () => {},
    selectedCategory: "someCategory",
    map: {
      on: () => {},
      off: () => {},
      getCenter: () => {},
    },
    places: {},
    router: {},
    showNewPin: () => {},
  };
  test("renders input form", () => {
    const wrapper = shallow(<InputForm {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("renders form validation errors", () => {
    const wrapper = shallow(<InputForm {...defaultProps} />);
    wrapper.setState({
      formValidationErrors: new Set(["asdf", "asdf2"]),
    });

    expect(wrapper.find(WarningMessagesContainer)).toHaveLength(1);
  });

  test("renders form fields", () => {
    const wrapper = shallow(<InputForm {...defaultProps} />);
    expect(wrapper.find(FormField)).toHaveLength(2);
  });

  test("onSubmit creates form validation errors", () => {
    const fieldName1 = "test1";
    const fieldName2 = "test2";
    const fieldName3 = "test3";
    const props = {
      ...defaultProps,
      selectedCategoryConfig: {
        fields: [
          { name: fieldName1, [constants.FIELD_VALIDITY_KEY]: "error1" },
          { name: fieldName2 },
          { name: fieldName3, [constants.FIELD_VALIDITY_KEY]: "error3" },
        ],
      },
    };

    const wrapper = shallow(<InputForm {...props} />);
    const errorKey1 = "error1";
    const errorKey2 = "error2";
    const errorKey3 = "error3";
    wrapper.setState({
      fields: OrderedMap({
        [fieldName1]: Map({
          [constants.FIELD_RENDER_KEY]: categoryName + fieldName1,
          [constants.FIELD_VALUE_KEY]: "",
          [constants.FIELD_VALIDITY_KEY]: true,
          [constants.FIELD_VALIDITY_MESSAGE_KEY]: errorKey1,
        }),
        [fieldName2]: Map({
          [constants.FIELD_RENDER_KEY]: categoryName + fieldName2,
          [constants.FIELD_VALUE_KEY]: "",
          [constants.FIELD_VALIDITY_KEY]: false,
          [constants.FIELD_VALIDITY_MESSAGE_KEY]: errorKey2,
        }),
        [fieldName3]: Map({
          [constants.FIELD_RENDER_KEY]: categoryName + fieldName3,
          [constants.FIELD_VALUE_KEY]: "",
          [constants.FIELD_VALIDITY_KEY]: false,
          [constants.FIELD_VALIDITY_MESSAGE_KEY]: errorKey3,
        }),
      }),
      updatingField: null,
      isFormSubmitting: false,
      formValidationErrors: new Set(),
      showValidityStatus: false,
      isMapPositioned: false,
    });

    wrapper.instance().onSubmit(eventStub);
    expect(wrapper.state("formValidationErrors")).toEqual(
      new Set([errorKey2, errorKey3]),
    );
  });
});
