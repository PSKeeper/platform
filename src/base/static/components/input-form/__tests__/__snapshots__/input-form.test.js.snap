// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm renders input form 1`] = `
<div
  className="input-form"
>
  <form
    className="input-form__form"
    id="mapseed-input-form"
    onSubmit={[Function]}
  >
    <FormField
      attachmentModels={Immutable.List []}
      categoryConfig={
        Object {
          "category": "someCategory",
          "fields": Array [
            Object {
              "name": "test1",
            },
            Object {
              "name": "test2",
            },
          ],
        }
      }
      disabled={false}
      fieldConfig={
        Object {
          "name": "test1",
        }
      }
      fieldState={
        Immutable.Map {
          "value": "",
          "renderKey": "someCategorytest1",
        }
      }
      key="someCategorytest1"
      map={
        Object {
          "getCenter": [Function],
          "off": [Function],
          "on": [Function],
        }
      }
      mapConfig={Object {}}
      onAddAttachment={[Function]}
      onFieldChange={[Function]}
      onGeometryStyleChange={[Function]}
      places={Object {}}
      router={Object {}}
      showValidityStatus={false}
      t={[Function]}
      updatingField={null}
    />
    <FormField
      attachmentModels={Immutable.List []}
      categoryConfig={
        Object {
          "category": "someCategory",
          "fields": Array [
            Object {
              "name": "test1",
            },
            Object {
              "name": "test2",
            },
          ],
        }
      }
      disabled={false}
      fieldConfig={
        Object {
          "name": "test2",
        }
      }
      fieldState={
        Immutable.Map {
          "value": "",
          "renderKey": "someCategorytest2",
        }
      }
      key="someCategorytest2"
      map={
        Object {
          "getCenter": [Function],
          "off": [Function],
          "on": [Function],
        }
      }
      mapConfig={Object {}}
      onAddAttachment={[Function]}
      onFieldChange={[Function]}
      onGeometryStyleChange={[Function]}
      places={Object {}}
      router={Object {}}
      showValidityStatus={false}
      t={[Function]}
      updatingField={null}
    />
  </form>
</div>
`;
